bids/grabbids/bids_layout.py

```diff
@@ -88,16 +84,25 @@ def map_conf(x):
         for i, p in enumerate(paths):
             if isinstance(p, six.string_types):
                 paths[i] = (p, conf_path % 'bids')
-                if len(paths) == 1 and root is None:
-                    root = p
             elif isinstance(p, tuple):
                 doms = [map_conf(d) for d in listify(p[1])]
                 paths[i] = (p[0], doms)
 
-        self.root = '/' if root is None else root
-
-        target = pathjoin(self.root, 'dataset_description.json')
-        if not exists(target):
+        # Set root to longest valid common parent if it isn't explicitly set
+        if root is None:
+            abs_paths = [os.path.abspath(p[0]) for p in paths]
+            root = os.path.commonprefix(abs_paths)
+            if not root:
+                raise ValueError("One or more invalid paths passed; could not "
+                                 "find a common parent directory of %s." %
+                                 abs_paths)
+            elif not os.path.isdir(root):
```

Comment:
Python 3.5 fixed this issue with `commonpath`. It might be cleaner to handle this edge case by using that and providing a fall-back implementation of `commonpath`:

```Python
try:
    from os.path import commonpath
except ImportError:
    def commonpath(paths):
        prefix = os.path.commonprefix(paths)
        if not os.path.isdir(prefix):
            prefix = os.path.dirname(prefix)
        return prefix
```

An additional concern: This default root finding doesn't allow Windows users to have some data on one drive and some on another, e.g. `['C:\some\dir', D:\other\dir]`. Is this a problem?

Reply:
I was initially using `commonpath`, and didn't realize it wasn't available on 2 until the tests failed. So then I switched to `commonprefix`. But I like the idea of trying for the former first, thanks.

Re: windows, good point, I hadn't thought of that. But I kind of lean towards letting it fail and forcing windows users to explicitly set a root. Seems preferable to assuming the first path in the list is the one the user wants.